mne_bids/path.py

```diff
@@ -571,9 +583,16 @@ def update(self, *, check=None, **kwargs):
 
         # error check entities
         for key, val in kwargs.items():
-            # check if there are any characters not allowed
-            if val is not None and key != 'root':
-                _check_key_val(key, val)
+            # XXX: suffix can be overridden with non-BIDS characters
+            # which allows `dataset_description.json` to be represented
+            if key != 'suffix':
+                # check if there are any characters not allowed
+                if val is not None and key != 'root':
+                    _check_key_val(key, val)
+            elif self.check:
+                # check if there are any characters not allowed
+                if val is not None and key != 'root':
+                    _check_key_val(key, val)
```

Comment:
Isn't this equivalent?

```suggestion
            if key == 'suffix' and not self.check:
                continue
            if val is not None and key != 'root':
                _check_key_val(key, val)
```

Looks much simpler to me